torch/nn/parallel/distributed.py

```diff
@@ -388,6 +393,14 @@ def __init__(
             "doesn't have any parameter that requires a gradient."
         )
 
+        assert (
```

Comment:
Probably better to raise `ValueError` as I mentioned above. Looks like in other places we use `assert`, but I feel raising ValueError is more appropriate for invalid arguments. We could evaluate replacing the other asserts with ValueError as well in a separate PR. Although, it might not be possible to do that from a backward compatibility point of view since some user might be catching AssertionError and doing some error handling and if we change it to ValueError, it breaks their code. This is unlikely but possible.

Reply:
Agree that `ValueError` is better in this case. Previously I tried to keep the consistency with the subsequent assertions. Changed to `ValueError`

Will replace other `AssertionError` with `ValueError` in a separate PR.